From 0f922a7c1066321c17e2abbfe8454799a3b1bf5f Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 07 Aug 2022 16:19:55 +0300
Subject: [PATCH] [SW-99391] gaudi3: change condition to skip d2d init

As per efuse documentation, D2D_DIS bit is defined as below
0: do not perform D2D init
1: Perform D2D init

In ROM code, the bit was used opposite to what defined hence fix it.

Change-Id: I04e150e4d9f6914269f2413c85ce8a7ac2d96342
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 63e26e5..b714f9c 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -150,7 +150,7 @@
 	if (arcboot_rom_global.sku)
 		return;
 
-	if (arcboot_rom_global.config2_fuse & EFUSE_SYSCFG2_D2D_DIS_MASK)
+	if (!(arcboot_rom_global.config2_fuse & EFUSE_SYSCFG2_D2D_DIS_MASK))
 		return; /* skip d2d init */
 
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_D2D_LINK_START);
